garrysmod/lua/includes/modules/killicon.lua

```diff
@@ -81,7 +95,15 @@ function GetSize( name )
 		h = h * 0.75
 		
 		-- Make h/w 1:1
-		local tw, th = surface.GetTextureSize( t.texture )
+		local tw, th
+		
+		if ( t.type == TYPE_TEXTURE ) then
+			tw, th = surface.GetTextureSize( t.texture )
+		elseif ( t.type == TYPE_MATERIAL ) then
+			tw = t.w
+			th = t.h
+		end
+		

```

Comment:
This code causes the resulting icon to render bigger than specified in the `AddUV` function.

Tested using this:
```
killicon.AddUV( "monster_alien_slave", "hud/dneg_images", color_white, 193, 449, 60, 30 )
killicon.AddUV( "worldspawn", "killicons.png", color_white, 193, 449, 60, 30 )
```
killicons.png is the file linked in the OP.

```
if ( t.type == TYPE_MATERIAL ) then
	w = t.w
	h = t.h
end
```
Something like this works better (instead of `if ( t.type == TYPE_TEXTURE || t.type == TYPE_MATERIAL ) then`)

The default killicon rendering in Base gamemode has its own vertical alignment issues that I will need to look into resolving.
Kinda sad that the killicons offset themselves for no reason in the `killicon.Draw` function, instead of doing that in the base gamemode.

Reply:
In hindsight a more appropriate name for this func would be `AddTexCoord`

Reply:
> This code causes the resulting icon to render bigger than specified in the `AddUV` function.

This is how killicon.Add would handle a 60x30 texture too, though, and I can't imagine someone specifying a size of 512x512, for example, would actually want it drawn at that size.

`killicon.Add( "worldspawn", mat:GetName(), color_white )`
`killicon.AddTexCoord( "worldspawn", mat:GetName(), color_white, 0, 0, mat:Width(), mat:Height() )`

These two produce the same result, are you sure they shouldn't? If so I'll change it.